xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -42,10 +42,48 @@
 #include "utils/log.h"
 #include "utils/StringUtils.h"
 #include "WinEventsWayland.h"
+#include "utils/MathUtils.h"
 
 using namespace KODI::WINDOWING::WAYLAND;
 using namespace std::placeholders;
 
+namespace
+{
+
+// Caller should hold g_graphicsContext lock
```

Comment:
Have to lock because the resolutions might get updated while we are running through the loop. But I guess it's better to just add a lock instead of the comment :-)